### ISSUE TITLE
Add Request Limitation for Embedding Models

### ISSUE THREAD

**Issue by 2slow4u** [logancyang/obsidian-copilot#633]
**Is your feature request related to a problem? Please describe.**
When trying to index and embedd a large vault through the selected embedding model - in my case Azure OpenAI - I run into HTTP 429 Errors, which are indicating that the Rate Limit kicked in on the embedding model. I maxed out the Quota on Azure OpenAI but still getting it.

**Describe the solution you'd like**
In order to avoid HTTP 429 Errors, it would be great to introduce an option to rate limit the number of request for the embedding creation.

**Describe alternatives you've considered**
Additionally there may be an option that influences the ramp up speed to the max limit of embedding requests in order to warm up the model and avoid through this the HTTP 429 Error.

**Additional context**
It would be great to extend the configurability on the vector store, e.g. chunking size, overlap, etc. That helps to experiment a little bit with the retrieval quality.

**Comment by logancyang** [comment]
Right, a setting that throttles the embedding request is needed.

**Comment by logancyang** [comment]
Thanks for reporting this, and sorry for the slow reply.

Copilot v3 is in code freeze, so we're closing this as not planned. v4 is a ground-up rewrite and is launching soon, with a very different architecture around agent mode.

If this is still an issue for you in v4 once it lands, please open a new issue and we'll take a fresh look. 🙏
